### PR TITLE
feat(migrations): Add support for optional migration groups

### PR DESCRIPTION
**Pull request by lynnagara**
Allows migration groups to be marked as optional and disabled via
settings.

The migration groups that are currently allowed to be optional are:
transactions, outcomes, sessions and querylog.

Only querylog is turned off by default.

### CODE REVIEW

File: snuba/migrations/groups.py [hunk 1]
```diff
@@ -15,6 +16,23 @@ class MigrationGroup(Enum):
     QUERYLOG = "querylog"
 
 
+# Migration groups are mandatory by default, unless they are on this list
+OPTIONAL_GROUPS = {
+    MigrationGroup.TRANSACTIONS,
+    MigrationGroup.OUTCOMES,
+    MigrationGroup.SESSIONS,
```
Comment (fpacifici) [inline snuba/migrations/groups.py:23]: I am not sure I would make these optional. The on prem standard install of sentry strictly depends on them.

**Review by fpacifici — APPROVED** [review]
Seems ok to me, but I would remove some of those groups from the optional list

File: snuba/migrations/groups.py (hunk 1 above)
Comment (lynnagara) [reply snuba/migrations/groups.py:23]: I'm ok with that. Thoughts on just leaving sessions on this list together with querylog? Reason being that single tenant does not run sessions so it'd be nice to skip the migrations.

Comment (fpacifici) [reply snuba/migrations/groups.py:23]: ok

**Pull request merged**